Test/NotCSharp/Python/EASY-N-AryTreePreorderTraversal.py

```diff
@@ -0,0 +1,22 @@
+#https://leetcode.com/problems/n-ary-tree-preorder-traversal/
+
+"""
+# Definition for a Node.
+class Node:
+    def __init__(self, val, children):
+        self.val = val
+        self.children = children
+"""
+class Solution:
+    def preorder(self, root: 'Node') -> List[int]:
+        if not root:
+            return []
+
+        lst = []
+        node = [root]
+
+        while node:
+            curr_node = node.pop(0)
+            lst.append(curr_node.val)
+            node = curr_node.children + node
```

Comment:
does this ordering matter? if so, why.
This is critical to this question and we must fully understand this.

Here is something similar I was doing this morning, that was not working.
https://paste.ofcode.org/f5V8iJsmkv6evLS43upsBE

Can you help investigate why it may be wrong

Reply:
here's an updated version of this:
https://paste.ofcode.org/7neAUAAjSA5X3wD2HrAgp6

Reply:
it is failing for the base test case, we see
                     1
                   /   \    \
                 3    2    4
                / \
              5   6

actual answer from above: [1,4,2,3,6,5]
expected: [1,3,5,6,2,4]

Reply:
Sorry for the delayed response. Could you please paste the paste of code link again. It seems to have expired. Thanks!

Reply:
Please see my latest commit. It does the same thing as before but is clearer and more intuitive this way. Also see the image below as reference for a test case code walk-through.
I am still happy to do a root cause for your code whenever you are able to post it. Thanks.
![20190927_103349](https://user-images.githubusercontent.com/14189245/65789627-a2e3a680-e112-11e9-90a6-6d86e6110109.jpg)